File: web/lib/raw-runner-command.test.ts
import assert from "node:assert/strict";
import { isRawRunnerCommand } from "./raw-runner-command";

const cases: Array<[string, string | null | undefined, boolean]> = [
  ["slash command", "/goal ship this", true],
  ["leading whitespace", "  /exit", true],
  ["plain text with slash later", "please run /goal", false],
  ["empty", "", false],
  ["undefined", undefined, false],
];

for (const [name, input, want] of cases) {
  assert.equal(isRawRunnerCommand(input), want, name);
  console.log(`ok ${name}`);
}

File: desktop/agent/raw_runner_command_test.go
package main

import "testing"

func TestRawRunnerCommandClassifier(t *testing.T) {
	cases := []struct {
		name  string
		input string
		want  bool
	}{
		{"goal", "/goal ship it", true},
		{"leading whitespace", " \n\t/exit", true},
		{"slash later is prose", "please run /goal", false},
		{"empty", "", false},
	}
	for _, tc := range cases {
		if got := isRawRunnerCommand(tc.input); got != tc.want {
			t.Fatalf("%s: isRawRunnerCommand(%q) = %v, want %v", tc.name, tc.input, got, tc.want)
		}
	}
}

func TestRawRunnerPromptPrefersLatestUserSlashCommand(t *testing.T) {
	task := &Task{
		Title:       "title",
		Description: "wrapped description",
		Turns: []ConversationTurn{
			{Role: "user", Content: "normal prompt"},
			{Role: "assistant", Content: "ok"},
			{Role: "user", Content: "  /exit"},
		},
	}
	if got := rawRunnerPromptForTask(task, "fallback"); got != "/exit" {
		t.Fatalf("rawRunnerPromptForTask = %q, want /exit", got)
	}
}
